cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -140,20 +201,25 @@ private String generateConfigFileName(String appName) {
    * @param appRelativePath   the relative {@link Path} of the application to write to
    * @return A valid application config file relative path
    */
-  private Path validateAppConfigRelativePath(RepositoryManager repositoryManager,
-                                             Path appRelativePath)
-    throws PushFailureException {
+  private Path validateAppConfigRelativePath(RepositoryManager repositoryManager, Path appRelativePath) throws
+    IllegalArgumentException {
     Path filePath = repositoryManager.getRepositoryRoot().resolve(appRelativePath);
     if (Files.isSymbolicLink(filePath)) {
-      throw new PushFailureException(String.format("Cannot push %s to the remote repository because it already " +
-                                                     "exists as a symbolic link. Symbolic links are not allowed.",
-                                                   appRelativePath));
+      throw new IllegalArgumentException(String.format(
+        "%s exists but refers to a symbolic link. Symbolic links are " + "not allowed.", appRelativePath));
     }
     if (Files.isDirectory(filePath)) {
-      throw new PushFailureException(String.format("Cannot push %s to the remote repository because it already " +
-                                                     "exists as a directory.", appRelativePath));
+      throw new IllegalArgumentException(String.format("%s refers to a directory not a file.", appRelativePath));
     }
     return filePath;
   }
+
+  private void closeRepositoryMangerQuietly(RepositoryManager repositoryManager) {
```

Comment:
Maybe we could mute the IOException in RepoManager? https://github.com/cdapio/cdap/pull/14958/commits/f442231cede421fb7cb5d437ba53814788eb1aa4

Reply:
Done.